tests/testthat/test-app_pws.R

```diff
@@ -0,0 +1,300 @@
+      vals$output$`plot_with_settings-plot_main`$src,
+      fixed = TRUE
+    )
+  )
+  # nolint end
+})
+
+# download plots. expect_download() might not be stable, hence we test
+# setting inputs and plot name changes
+testthat::test_that("plot_with_settings: download functionality ggplot2", {
+  skip_on_cran()
+  skip_on_ci()
+
+  # test default download options
+  app$click("plot_with_settings-downbutton-downl")
+  date <- strftime(Sys.time(), format = "%Y%m%d")
+  testthat::expect_true(
+    grepl(
+      paste0("plot_", date),
+      app$get_value(input = "plot_with_settings-downbutton-file_name")
+    )
+  )
+  testthat::expect_equal(
+    app$get_value(input = "plot_with_settings-downbutton-file_format"), "png"
+  )
+
+  # Download svg format
+  app$set_inputs(`plot_with_settings-downbutton-downl_state` = TRUE)
+  app$set_inputs(`plot_with_settings-downbutton-file_format` = "svg")
+  app$set_inputs(`plot_with_settings-downbutton-file_name` = "plot_svg")
+  app$click("plot_with_settings-downbutton-downl")
+  testthat::expect_equal(
+    app$get_value(input = "plot_with_settings-downbutton-file_name"), "plot_svg"
+  )
+
+  # change plot width and height
+  app$click("plot_with_settings-expbut")
+  app$set_inputs(`plot_with_settings-width` = 300)
+  app$set_inputs(`plot_with_settings-height` = 300)
+  app$click("plot_with_settings-downbutton-downl")
+  vals <- app$get_values()
+
+  testthat::expect_equal(app$get_value(input = "plot_with_settings-downbutton-file_name"), "plot_svg")
+  testthat::expect_equal(isolate(vals$output$`plot_with_settings-plot_main`$height), 300)
+  testthat::expect_equal(isolate(vals$output$`plot_with_settings-plot_main`$width), 300)
+
+  # reset
+  app$set_inputs(`plot_with_settings-expbut_state` = FALSE)
+  app$set_inputs(`plot_with_settings-downbutton-downl_state` = FALSE)
+})
+
+# downloading plot with modal
+testthat::test_that("plot_with_settings: download ggplot2 modal", {
+  skip_on_cran()
+  skip_on_ci()
+
+  # default downloading with modal
+  app$set_inputs(
+    `plot_with_settings-plot_hover` = character(0),
+    allow_no_input_binding_ = TRUE
+  )
+  app$click("plot_with_settings-expand")
+  app$click("plot_with_settings-modal_downbutton-downl")
+  date <- strftime(Sys.time(), format = "%Y%m%d")
+  testthat::expect_true(
+    grepl(
+      paste0("plot_", date),
+      app$get_value(input = "plot_with_settings-modal_downbutton-file_name")
+    )
+  )
+
+  # Change width and height of plot on modal
+  app$click("plot_with_settings-expand")
+  app$click("plot_with_settings-modal_downbutton-downl")
+  app$set_inputs(`plot_with_settings-height_in_modal` = 400)
+  app$set_inputs(`plot_with_settings-width_in_modal` = 500)
+  app$set_inputs(`plot_with_settings-modal_downbutton-file_name` = "plot_pdf")
+  app$set_inputs(`plot_with_settings-modal_downbutton-file_format` = "pdf")
+  app$set_inputs(`plot_with_settings-height_in_modal` = 1200)
+  app$set_inputs(`plot_with_settings-width_in_modal` = 750)
```

Comment:
same inputs are changed multiple times. I'd suggest to set them once and optimally with the app$click